features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/AbstractSemanticsNodeMapper.kt

```diff
@@ -21,6 +21,10 @@ internal abstract class AbstractSemanticsNodeMapper(
     private val semanticsUtils: SemanticsUtils = SemanticsUtils()
 ) : SemanticsNodeMapper {
 
+    protected fun resolveId(semanticsNode: SemanticsNode, currentIndex: Int = 0): Long {
+        return semanticsNode.id.toLong() shl SEMANTICS_ID_BIT_SHIFT + currentIndex
```

Comment:
maybe it is worth adding a comment explaining why shift is needed - I guess it is to separate final ID into 2 chunks of 32 bits: ID for the node and current index?